AUTHORING_GUIDE.md

```diff
@@ -347,8 +348,43 @@ is expected. Strive to verify the content of the output rather than the syntax.
 For example, the test might verify that a string is included in the output,
 without taking a dependency on where that string occurs in the output.
 
-### Running tests
+### Avoid infinite loops
+
+Never put potential infinite loops in the test code path. A typical
+example is about gRPC's LongRunningOperations. Make sure you pass the
+timeout parameter to the `result()` call.
+
+Good:
+
+```python
+# will raise google.api_core.GoogleAPICallError after 60 seconds
+operation.result(60)
+```
+
+Bad:
+
+```python
+operation.result()  # this could wait forever.
+```
 
+We recommend the timeout parameter to be around the number that gives
+you more than 90% success rate. Don't put too long a timeout.
+
+Now this test is inevitably flaky, so consider marking the test as
+`flaky` as follows:
+
+```python
+
+@pytest.mark.flaky(max_runs=3, min_passes=1)
```

Comment:
Wasn't there an issue with pytest and flaky? Do we want to recommend backoff instead?

Reply:
What was the issue? I heard this several times, but I'm not sure what the issue was.

Reply:
`flaky` gives you a test report, something like:

```
===Flaky Test Report===

test_counter failed (9 runs remaining out of 10).
        <class 'Exception'>
        not yet
        [<TracebackEntry /usr/local/google/home/tmatsuo/work/tmp/test.py:12>]
test_counter failed (8 runs remaining out of 10).
        <class 'Exception'>
        not yet
        [<TracebackEntry /usr/local/google/home/tmatsuo/work/tmp/test.py:12>]
test_counter passed 1 out of the required 1 times. Success!

===End Flaky Test Report===

```

I think it's useful when we want to scan these logs.

Reply:
I thought you had mentioned it, but if it's working well ignore me :)